File: backend/src/main/java/com/example/questifyapp/dto/learningUnit/LearningUnitDto.java
package com.example.questifyapp.dto.learningUnit;

import com.example.questifyapp.dto.LearningUnitTypeDto;
import com.example.questifyapp.dto.UserDto;
import com.example.questifyapp.dto.exercise.ExerciseResponseDto;

import java.time.LocalDateTime;
import java.util.List;
public record LearningUnitDto(
        Long id,
        String name,
        String code,
        String description,
        LearningUnitTypeDto type,
        int status,
        LocalDateTime createdAt,
        UserDto createdBy,
        Long parentId,
        List<LearningUnitChildDto> childUnits,
        List<ExerciseResponseDto> exercises,
        Long numberOfExercise
) {}

File: frontend/src/model/LearningUnitDTO.ts
import { LearningUnitChildDto } from "./LearningUnitChildDto";
import { LearningUnitType } from "./LearningUnitType";
import { ExerciseDTO } from "./ExerciseDTO";
import {LessonConfigDTO} from "./LessonConfigDTO";

export interface LearningUnitDTO {
  id: string;
  name: string;
  code: string;
  description: string;
  type: LearningUnitType;
  status: number;
  createdAt: Date;
  createdBy: string;
  updatedAt?: string;
  children?: LearningUnitChildDto[];
  exerciseCategories?: LearningUnitChildDto[];
  exercises?: ExerciseDTO[];
  lessonConfig?: LessonConfigDTO;
  parentId: string;
  numberOfComplete: number;
  numberOfExercise: number;
}

export interface CourseDTO {
  id: string;
  name: string;
  code: string;
  status: string;
  description: string;
  createdAt?: string;
  updatedAt?: string;
  numberOfComplete: number;
  numberOfExercise: number;
}

export interface LearningUnitWithChildren {
  id: string;
  name: string;
  code: string;
  description: string;
  type: LearningUnitType;
  status: number;
  createdAt: Date;
  createdBy: string;
  children: LearningUnitWithChildren[];
  numberOfExercises: number;
  updatedAt?: string;
  parentId: string;
}

export interface LearningUnitWithLessonConfig{
  id: string;
  name: string;
  code: string;
  description: string;
  type: LearningUnitType;
  status: number;
  createdAt: Date;
  createdBy: string;
  updatedAt?: string;
  exerciseCategories?: LearningUnitChildDto[];
  lessonConfig?: LessonConfigDTO;
  parentId: string;
}